### PR TITLE
Remove updates from pod mutating webhook

### PR DESCRIPTION
**Pull request by andrewdinunzio**
**Description:** <Describe what has changed.>

The opentelemetry operator sometimes modifies pods in an invalid way. I've only ever experienced this happening on UPDATE events, and since it's not necessary to update pods, this change removes UPDATE from the webhook configs and leaves only CREATE.

**Link to tracking Issue:** Resolves: open-telemetry/opentelemetry-operator#1514

**Testing:** No tests added.

**Documentation:** No documentation added.

### CODE REVIEW

**Comment by andrewdinunzio** [comment]
e2e tests passed, but some unit test failures:
```
--- FAIL: TestReplaceConfig (0.01s)
    --- FAIL: TestReplaceConfig/should_rewrite_scrape_configs_with_SD_config_when_TargetAllocator_is_enabled_and_feature_flag_is_not_set (0.01s)
        config_replace_test.go:188: 
                Error Trace:    /home/andrew/coding/forks/opentelemetry-operator/internal/manifests/collector/config_replace_test.go:188
                Error:          Not equal: 
                                expected: "debug: null\r\nexporters: null\r\nprocessors: null\r\nreceivers:\r\n  prometheus:\r\n    config:\r\n      global:\r\n        evaluation_interval: 1m\r\n        scrape_interval: 1m\r\n        scrape_timeout: 10s\r\n      scrape_configs:\r\n      - honor_labels: true\r\n        http_sd_configs:\r\n        - url: http://test-targetallocator:80/jobs/service-x/targets?collector_id=$POD_NAME\r\n        job_name: service-x\r\n        metric_relabel_configs:\r\n        - action: keep\r\n          regex: (.*)\r\n          separator: ;\r\n          source_labels:\r\n          - label1\r\n        - action: labelmap\r\n          regex: (.*)\r\n          separator: ;\r\n          source_labels:\r\n          - label4\r\n        metrics_path: /metrics\r\n        relabel_configs:\r\n        - action: keep\r\n          regex: (.*)\r\n          source_labels:\r\n          - label1\r\n        - action: replace\r\n          regex: (.*)\r\n          replacement: $$1_$$2\r\n          separator: ;\r\n          source_labels:\r\n          - label2\r\n          target_label: label3\r\n        - action: labelmap\r\n          regex: (.*)\r\n          separator: ;\r\n          source_labels:\r\n          - label4\r\n        - action: labeldrop\r\n          regex: foo_.*\r\n        scheme: http\r\n        scrape_interval: 1m\r\n        scrape_timeout: 10s\r\nservice:\r\n  pipelines:\r\n    metrics:\r\n      exporters:\r\n      - debug\r\n      processors: []\r\n      receivers:\r\n      - prometheus\r\n"
                                actual  : "debug: null\nexporters: null\nprocessors: null\nreceivers:\n  prometheus:\n    config:\n      global:\n        evaluation_interval: 1m\n        scrape_interval: 1m\n        scrape_timeout: 10s\n      scrape_configs:\n      - honor_labels: true\n        http_sd_configs:\n        - url: http://test-targetallocator:80/jobs/service-x/targets?collector_id=$POD_NAME\n        job_name: service-x\n        metric_relabel_configs:\n        - action: keep\n          regex: (.*)\n          separator: ;\n          source_labels:\n          - label1\n        - action: labelmap\n          regex: (.*)\n          separator: ;\n          source_labels:\n          - label4\n        metrics_path: /metrics\n        relabel_configs:\n        - action: keep\n          regex: (.*)\n          source_labels:\n          - label1\n        - action: replace\n          regex: (.*)\n          replacement: $$1_$$2\n          separator: ;\n          source_labels:\n          - label2\n          target_label: label3\n        - action: labelmap\n          regex: (.*)\n          separator: ;\n          source_labels:\n          - label4\n        - action: labeldrop\n          regex: foo_.*\n        scheme: http\n        scrape_interval: 1m\n        scrape_timeout: 10s\nservice:\n  pipelines:\n    metrics:\n      exporters:\n      - debug\n      processors: []\n      receivers:\n      - prometheus\n"
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1,58 +1,58 @@
                                -debug: null
                                -exporters: null
                                -processors: null
                                -receivers:
                                -  prometheus:
                                -    config:
                                -      global:
                                -        evaluation_interval: 1m
                                -        scrape_interval: 1m
                                -        scrape_timeout: 10s
                                -      scrape_configs:
                                -      - honor_labels: true
                                -        http_sd_configs:
                                -        - url: http://test-targetallocator:80/jobs/service-x/targets?collector_id=$POD_NAME
                                -        job_name: service-x
                                -        metric_relabel_configs:
                                -        - action: keep
                                -          regex: (.*)
                                -          separator: ;
                                -          source_labels:
                                -          - label1
                                -        - action: labelmap
                                -          regex: (.*)
                                -          separator: ;
                                -          source_labels:
                                -          - label4
                                -        metrics_path: /metrics
                                -        relabel_configs:
                                -        - action: keep
                                -          regex: (.*)
                                -          source_labels:
                                -          - label1
                                -        - action: replace
                                -          regex: (.*)
                                -          replacement: $$1_$$2
                                -          separator: ;
                                -          source_labels:
                                -          - label2
                                -          target_label: label3
                                -        - action: labelmap
                                -          regex: (.*)
                                -          separator: ;
                                -          source_labels:
                                -          - label4
                                -        - action: labeldrop
                                -          regex: foo_.*
                                -        scheme: http
                                -        scrape_interval: 1m
                                -        scrape_timeout: 10s
                                -service:
                                -  pipelines:
                                -    metrics:
                                -      exporters:
                                -      - debug
                                -      processors: []
                                -      receivers:
                                -      - prometheus
                                +debug: null
                                +exporters: null
                                +processors: null
                                +receivers:
                                +  prometheus:
                                +    config:
                                +      global:
                                +        evaluation_interval: 1m
                                +        scrape_interval: 1m
                                +        scrape_timeout: 10s
                                +      scrape_configs:
                                +      - honor_labels: true
                                +        http_sd_configs:
                                +        - url: http://test-targetallocator:80/jobs/service-x/targets?collector_id=$POD_NAME
                                +        job_name: service-x
                                +        metric_relabel_configs:
                                +        - action: keep
                                +          regex: (.*)
                                +          separator: ;
                                +          source_labels:
                                +          - label1
                                +        - action: labelmap
                                +          regex: (.*)
                                +          separator: ;
                                +          source_labels:
                                +          - label4
                                +        metrics_path: /metrics
                                +        relabel_configs:
                                +        - action: keep
                                +          regex: (.*)
                                +          source_labels:
                                +          - label1
                                +        - action: replace
                                +          regex: (.*)
                                +          replacement: $$1_$$2
                                +          separator: ;
                                +          source_labels:
                                +          - label2
                                +          target_label: label3
                                +        - action: labelmap
                                +          regex: (.*)
                                +          separator: ;
                                +          source_labels:
                                +          - label4
                                +        - action: labeldrop
                                +          regex: foo_.*
                                +        scheme: http
                                +        scrape_interval: 1m
                                +        scrape_timeout: 10s
                                +service:
                                +  pipelines:
                                +    metrics:
                                +      exporters:
                                +      - debug
                                +      processors: []
                                +      receivers:
                                +      - prometheus
                                 
                Test:           TestReplaceConfig/should_rewrite_scrape_configs_with_SD_config_when_TargetAllocator_is_enabled_and_feature_flag_is_not_set
    --- FAIL: TestReplaceConfig/should_remove_scrape_configs_if_TargetAllocator_is_enabled_and_feature_flag_is_set (0.00s)
        config_replace_test.go:201: 
                Error Trace:    /home/andrew/coding/forks/opentelemetry-operator/internal/manifests/collector/config_replace_test.go:201
                Error:          Not equal: 
                                expected: "debug: null\r\nexporters: null\r\nprocessors: null\r\nreceivers:\r\n  prometheus:\r\n    config:\r\n      global:\r\n        evaluation_interval: 1m\r\n        scrape_interval: 1m\r\n        scrape_timeout: 10s\r\n    target_allocator:\r\n      collector_id: ${POD_NAME}\r\n      endpoint: http://test-targetallocator:80\r\n      interval: 30s\r\nservice:\r\n  pipelines:\r\n    metrics:\r\n      exporters:\r\n      - debug\r\n      processors: []\r\n      receivers:\r\n      - prometheus\r\n"
                                actual  : "debug: null\nexporters: null\nprocessors: null\nreceivers:\n  prometheus:\n    config:\n      global:\n        evaluation_interval: 1m\n        scrape_interval: 1m\n        scrape_timeout: 10s\n    target_allocator:\n      collector_id: ${POD_NAME}\n      endpoint: http://test-targetallocator:80\n      interval: 30s\nservice:\n  pipelines:\n    metrics:\n      exporters:\n      - debug\n      processors: []\n      receivers:\n      - prometheus\n"
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1,23 +1,23 @@
                                -debug: null
                                -exporters: null
                                -processors: null
                                -receivers:
                                -  prometheus:
                                -    config:
                                -      global:
                                -        evaluation_interval: 1m
                                -        scrape_interval: 1m
                                -        scrape_timeout: 10s
                                -    target_allocator:
                                -      collector_id: ${POD_NAME}
                                -      endpoint: http://test-targetallocator:80
                                -      interval: 30s
                                -service:
                                -  pipelines:
                                -    metrics:
                                -      exporters:
                                -      - debug
                                -      processors: []
                                -      receivers:
                                -      - prometheus
                                +debug: null
                                +exporters: null
                                +processors: null
                                +receivers:
                                +  prometheus:
                                +    config:
                                +      global:
                                +        evaluation_interval: 1m
                                +        scrape_interval: 1m
                                +        scrape_timeout: 10s
                                +    target_allocator:
                                +      collector_id: ${POD_NAME}
                                +      endpoint: http://test-targetallocator:80
                                +      interval: 30s
                                +service:
                                +  pipelines:
                                +    metrics:
                                +      exporters:
                                +      - debug
                                +      processors: []
                                +      receivers:
                                +      - prometheus
                                 
                Test:           TestReplaceConfig/should_remove_scrape_configs_if_TargetAllocator_is_enabled_and_feature_flag_is_set
--- FAIL: TestDesiredConfigMap (0.01s)
    --- FAIL: TestDesiredConfigMap/should_return_expected_collector_config_map (0.00s)
        configmap_test.go:71: 
                Error Trace:    /home/andrew/coding/forks/opentelemetry-operator/internal/manifests/collector/configmap_test.go:71
                Error:          Not equal: 
                                expected: map[string]string{"collector.yaml":"processors:\nreceivers:\n  jaeger:\n    protocols:\n      grpc:\n  prometheus:\n    config:\n      scrape_configs:\n      - job_name: otel-collector\n        scrape_interval: 10s\n        static_configs:\n          - targets: [ '0.0.0.0:8888', '0.0.0.0:9999' ]\n\nexporters:\n  debug:\n\nservice:\n  pipelines:\n    metrics:\n      receivers: [prometheus, jaeger]\n      processors: []\n      exporters: [debug]"}
                                actual  : map[string]string{"collector.yaml":"processors:\r\nreceivers:\r\n  jaeger:\r\n    protocols:\r\n      grpc:\r\n  prometheus:\r\n    config:\r\n      scrape_configs:\r\n      - job_name: otel-collector\r\n        scrape_interval: 10s\r\n        static_configs:\r\n          - targets: [ '0.0.0.0:8888', '0.0.0.0:9999' ]\r\n\r\nexporters:\r\n  debug:\r\n\r\nservice:\r\n  pipelines:\r\n    metrics:\r\n      receivers: [prometheus, jaeger]\r\n      processors: []\r\n      exporters: [debug]"}
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1,3 +1,3 @@
                                 (map[string]string) (len=1) {
                                - (string) (len=14) "collector.yaml": (string) (len=390) "processors:\nreceivers:\n  jaeger:\n    protocols:\n      grpc:\n  prometheus:\n    config:\n      scrape_configs:\n      - job_name: otel-collector\n        scrape_interval: 10s\n        static_configs:\n          - targets: [ '0.0.0.0:8888', '0.0.0.0:9999' ]\n\nexporters:\n  debug:\n\nservice:\n  pipelines:\n    metrics:\n      receivers: [prometheus, jaeger]\n      processors: []\n      exporters: [debug]"
                                + (string) (len=14) "collector.yaml": (string) (len=411) "processors:\r\nreceivers:\r\n  jaeger:\r\n    protocols:\r\n      grpc:\r\n  prometheus:\r\n    config:\r\n      scrape_configs:\r\n      - job_name: otel-collector\r\n        scrape_interval: 10s\r\n        static_configs:\r\n          - targets: [ '0.0.0.0:8888', '0.0.0.0:9999' ]\r\n\r\nexporters:\r\n  debug:\r\n\r\nservice:\r\n  pipelines:\r\n    metrics:\r\n      receivers: [prometheus, jaeger]\r\n      processors: []\r\n      exporters: [debug]"
                                 }
                Test:           TestDesiredConfigMap/should_return_expected_collector_config_map
error1: couldn't parse the opentelemetry-collector configurationerror2: no receivers available as part of the configurationFAIL
FAIL    github.com/open-telemetry/opentelemetry-operator/internal/manifests/collector   0.293s
```
Will look into that...

**Comment by andrewdinunzio** [comment]
Nevermind, I just had to run `find . -name "*.yaml" | xargs dos2unix` and now all tests pass for me locally.

**Comment by swiatekm-sumo** [comment]
@andrewdinunzio could you add a changelog entry? You just need to run `make chlog-new` and fill out the newly created yaml file in `~/.chloggen`.

The change itself looks fine to me, but I'd like @jaronoff97 and @pavolloffay to weigh in as well, as it potentially has significant impact.

**Comment by andrewdinunzio** [comment]
> @andrewdinunzio could you add a changelog entry? You just need to run `make chlog-new` and fill out the newly created yaml file in `~/.chloggen`.

Added. I set it as a "bug_fix". I don't think it should be a breaking change, but I don't know this repo very well so need confirmation about that.

**Review by pavolloffay — APPROVED** [review]

**Comment by pavolloffay** [comment]
@andrewdinunzio could you please resolve conflicts?

**Comment by pavolloffay** [comment]
@andrewdinunzio would it as well resolve issues linked in https://github.com/open-telemetry/opentelemetry-operator/issues/1514#issuecomment-1555100871  ?

**Comment by pavolloffay** [comment]
Ping @andrewdinunzio the conflicts needs to be resolved

**Comment by andrewdinunzio** [comment]
Used the main branch in my fork so syncing it closed the PR, but will reopen once I make the updates. Doing that now.

**Comment by andrewdinunzio** [comment]
@pavolloffay if you could check it over once more I think it should be good to go now

**Review by jaronoff97 — APPROVED** [review]

**Comment by andrewdinunzio** [comment]
reopening

**Review by jaronoff97 — APPROVED** [review]

**Comment by andrewdinunzio** [comment]
looks like it's good to merge?

File: .chloggen/main.yaml
Comment (iblancasa) [inline .chloggen/main.yaml:1]: I think it would make sense to use a more meaningful name for this file.

**Review by iblancasa — APPROVED** [review]

File: .chloggen/main.yaml
Comment (andrewdinunzio) [reply .chloggen/main.yaml:1]: ah it must get the name from the branch and I just worked on main in my fork. Would something like `webhook-remove-pod-update.yaml` work?

Comment (andrewdinunzio) [reply .chloggen/main.yaml]: Done!

**Review by swiatekm-sumo — APPROVED** [review]

**Comment by andrewdinunzio** [comment]
Thanks! I don't have the power to merge :D is there anything this is waiting on?

**Comment by swiatekm-sumo** [comment]
Only waiting for @pavolloffay 's approval at this point.

**Comment by andrewdinunzio** [comment]
Gotcha, thanks. He approved, but after fixing the conflict I re-requested his approval

**Review by pavolloffay — APPROVED** [review]

**Pull request merged**
